### PR TITLE
fix: find correct permission

### PR DESCRIPTION
**Pull request by joshua-hancox**
attempt to fix: #1192 (closed)

Closes: https://github.com/integrations/terraform-provider-github/issues/1183

### CODE REVIEW

**Comment by SharpEdgeMarshall** [comment]
With this PR you are removing the support for custom role permissions, am I right? We need that

**Comment by SharpEdgeMarshall** [comment]
@joshua-hancox @kfcampbell any news? this is blocking us from adopting this provider

**Comment by Luis-3M** [comment]
We're also interested in having this fix available - quiet confusing to see TF plans that always report changes when using custom roles.

File: github/resource_github_repository_collaborator.go [hunk 1]
```diff
@@ -35,10 +35,11 @@ func resourceGithubRepositoryCollaborator() *schema.Resource {
...
 			"permission": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Default:  "push",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "push",
+				ValidateFunc: validateValueFunc([]string{"pull", "triage", "push", "maintain", "admin"}),
```
Comment (kfcampbell) [inline github/resource_github_repository_collaborator.go:42]: The addition of this `ValidateFunc` means that custom roles can no longer be used, correct?

**Comment by nickfloyd** [comment]
Hey @joshua-hancox I'd be glad to resolve those merge conflicts for you to move this along if you'd like.  I'll need push or maintainer access to [joshua-hancox:main](https://github.com/joshua-hancox/terraform-provider-github/tree/main), though so that we can get your fork synced with this repo.  Let us know if you'd like us to handle that or if you'd be up for doing it!

**Comment by joshua-hancox** [comment]
@kfcampbell I don't believe that was part of my change here, sorry it's been ages since I looked at this.

@nickfloyd I doubt I have any time to look at this any time soon, I have added you as collaborator please feel free to carry on.

**Comment by SharpEdgeMarshall** [comment]
From my understanding this PR https://github.com/integrations/terraform-provider-github/pull/1192 was good but didn't consider backwards compatibility.
May I propose to release a new major with those changes or add some kind of mapping to preserve old naming compatibility?

**Review by kfcampbell — APPROVED** [review]
I'll get this merged and released in the near future.

**Review by kfcampbell — CHANGES_REQUESTED** [review]
Ahh oops I should've looked at the checks before approving :facepalm: 

@joshua-hancox do you have any interest in fixing up the merge conflicts? If not, I can take a look at it at some point.

**Comment by kfcampbell** [comment]
Sounds good! I've added the "Status: Up for grabs" label so the community can jump on the submission if anybody can get to it sooner rather than later.

**Review by kfcampbell — CHANGES_REQUESTED** [review]
I don't feel comfortable merging this PR given the diff/changes here. The commit history includes commits that I've force-pushed away, and I'm scared of reintroducing [this incident](https://github.com/integrations/terraform-provider-github/issues/1236#issuecomment-1198622552) merging this PR as-is. 

I'm going to close this for the time being, and would be very receptive to reviewing PRs that cherry-pick relevant commits or reimplement the functionality.

**Comment by kfcampbell** [comment]
g

**Pull request closed without merging**